File: AndroidApp/src/main/java/com/smp/sharedkotlin/app/support/Mapper.kt
package com.smp.sharedkotlin.app.support


interface Mapper<From, To> {
    fun map(obj: From): To
    fun mapAll(objs: Iterable<From>): List<To>
    fun reverse(obj: To): From
    fun reverseAll(objs: Iterable<To>): List<From>
}

File: iOSApp/iOSApp/Support/Mapper.swift

protocol Mapper {
    associatedtype From
    associatedtype To
    
    func map(obj: From) -> To
    func mapAll(objs: [From]) -> [To]
    
    func reverse(obj: To) -> From
    func reverseAll(objs: [To]) -> [From]
}
